app/app.go

```diff
@@ -456,236 +514,129 @@ func New(
+	res.TimeoutInfo.TimeoutPropose = app.TimeoutPropose()
 
-	// Invoke load latest version for its side-effect of invoking baseapp.Init()
-	if err := app.LoadLatestVersion(); err != nil {
-		panic(fmt.Sprintf("loading latest version: %s", err.Error()))
-	}
+	return res, nil
 }
 
-// InitChainer is middleware that gets invoked part-way through the baseapp's InitChain invocation.
-func (app *App) InitChainer(ctx sdk.Context, req abci.RequestInitChain) abci.ResponseInitChain {
-	var genesisState GenesisState
-	if err := tmjson.Unmarshal(req.AppStateBytes, &genesisState); err != nil {
-		panic(err)
-	}
-	appVersion := req.ConsensusParams.Version.AppVersion
-	app.UpgradeKeeper.SetModuleVersionMap(ctx, app.manager.GetVersionMap(appVersion))
-	return app.manager.InitGenesis(ctx, app.appCodec, genesisState, appVersion)
+// DefaultGenesis returns the default genesis state
+func (app *App) DefaultGenesis() GenesisState {
+	return app.BasicManager.DefaultGenesis(app.encodingConfig.Codec)
 }
 
 // LoadHeight loads a particular height
 func (app *App) LoadHeight(height int64) error {
 	return app.LoadVersion(height)
 }
 
-// SupportedVersions returns all the state machines that the
-// application supports
-func (app *App) SupportedVersions() []uint64 {
-	return app.manager.SupportedVersions()
-}
-
-// versionedKeys returns a map from moduleName to KV store key for the given app
-// version.
-func (app *App) versionedKeys(appVersion uint64) map[string]*storetypes.KVStoreKey {
-	output := make(map[string]*storetypes.KVStoreKey)
-	if keys, exists := app.keyVersions[appVersion]; exists {
-		for _, moduleName := range keys {
-			if key, exists := app.keys[moduleName]; exists {
-				output[moduleName] = key
-			}
-		}
+// ModuleAccountAddrs returns all the app's module account addresses.
+func (app *App) ModuleAccountAddrs() map[string]bool {
+	modAccAddrs := make(map[string]bool)
+	for acc := range maccPerms {
+		modAccAddrs[authtypes.NewModuleAddress(acc).String()] = true
 	}
-	return output
-}
 
-// baseKeys returns the base keys that are mounted to every version
-func (app *App) baseKeys() map[string]*storetypes.KVStoreKey {
-	return map[string]*storetypes.KVStoreKey{
-		// we need to know the app version to know what stores to mount
-		// thus the paramstore must always be a store that is mounted
-		paramstypes.StoreKey: app.keys[paramstypes.StoreKey],
-	}
+	return modAccAddrs
 }
 
-// ModuleAccountAddrs returns all the app's module account addresses.
-func (app *App) ModuleAccountAddrs() map[string]bool {
+// BlockedAddresses returns all the app's blocked account addresses.
+func (app *App) BlockedAddresses() map[string]bool {
 	modAccAddrs := make(map[string]bool)
-	for acc := range maccPerms {
+	for acc := range app.ModuleAccountAddrs() {
 		modAccAddrs[authtypes.NewModuleAddress(acc).String()] = true
 	}
 
+	// allow the following addresses to receive funds
+	delete(modAccAddrs, authtypes.NewModuleAddress(govtypes.ModuleName).String())
```

Comment:
why?

Reply:
good question, I'm not 100% sure actually, maybe @julienrbrt can shed some light on this

Reply:
Hey! This is the context: https://github.com/cosmos/cosmos-sdk/pull/12852.
In short allowing gov to receive funds from anyone, as well of the community pool.
Tbh I forgot the exact context of why storing extra funds in gov at the first place.